resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala

```diff
@@ -107,6 +107,9 @@ private[mesos] class MesosSubmitRequestServlet(
     val driverCores = sparkProperties.get("spark.driver.cores")
     val name = request.sparkProperties.getOrElse("spark.app.name", mainClass)
 
+    validateLabelFormat(sparkProperties, "spark.mesos.network.labels", "spark.mesos.task.labels",
```

Comment:
is this a finite list? If you know ahead of time what the list of labels to validate against are, I think you should move it inside the function so all you pass is the `sparkProperties` map e.g.
```
validateLabelFormat(sparkProperties)

Reply:
changed as suggested